src/main/cpp/blaze.cc

```diff
@@ -1424,14 +1424,28 @@ static map<string, EnvVarValue> PrepareEnvironmentForJvm() {
   if (!blaze::GetEnv("http_proxy").empty() ||
       !blaze::GetEnv("HTTP_PROXY").empty()) {
     BAZEL_LOG(WARNING)
-        << "detected http_proxy set in env, setting no_proxy for localhost.";
+        << "detected HTTP_PROXY set in env, setting NO_PROXY for localhost.";
 
     // Disable HTTP proxies for localhost and any localhost-like address,
     // in case we (or gRPC, etc.) ever use one of these addresses.
-    std::string localhost_addresses = "localhost,127.0.0.1,0:0:0:0:0:0:0:1,::1";
+    string localhost_addresses = blaze::GetEnv("NO_PROXY");
+    if (localhost_addresses.empty()) {
+      localhost_addresses = blaze::GetEnv("no_proxy");
+    }
+    vector<string> localhost_addresses_vec{"localhost", "127.0.0.1", "0:0:0:0:0:0:0:1", "::1"};
+    for (auto const& s : localhost_addresses_vec) {
+      if (localhost_addresses.find(s) == string::npos) {
+        localhost_addresses += ",";
```

Comment:
Sorry. I missed this before. You need go guard against having a leading ',' if NO_PROXY is not set.
How about:
     if (!localhost_addresses.empty()) {
       localhost_addresses += ',';
    }